### DISCUSSION TITLE
Releasing SeisSol

### DISCUSSION THREAD

**Discussion by sebwolf-de** [SeisSol/SeisSol#656] [Ideas]
Hi, since the merge of #625 comes closer, we have to think about proper releasing.
@ravil-mobile's and my idea is to release a version `0.9.0`, which contains the latest `master` branch before merging #625. Immediately after merging #625, we release a `1.0.0-rc` release candidate version of SeisSol. Probably, we will get some feedback from users about some bugs, which we introduced in #625, but went unnoticed until now. As time evolves, we will fix these bugs and then after a few weeks when everything has been smoothed out, we can release `1.0.0`.
What do you think @baderml @AliceGabriel?

**Comment by Thomas-Ulrich** [comment]
I think it is a good idea. But I would call it directly 1.0.0, and then we can increase the counter at each PR.

**Reply by ravil-mobile** [reply]
I guess increasing the counter after each PR is going to be too much. I think we can increase the patch version after each significant bug fix;  the minor version after some essential contribution (for example, introducing the actor model) ; the major version after some significant change (for example, when the entire SeisSol is ported to c++)

**Reply by sebwolf-de** [reply]
I would suggest that we could make it dependent on the parameter files:
* If the SeisSol core is not changed (e.g. changes to the documentation or analysis scripts): do not increase  (e.g. #652)
* If the SeisSol core is changed for a bugfix, but a _user_ doesn't necessarily realize these changes: increase patch (e.g. #638)
* If a new feature is added to the parameter file: increase minor (e.g #644)
* If a feature is removed from the parameter file: increase major (e.g. #625)

**Reply by krenzland** [reply]
The only thing that makes sense imo is to follow semver which is basically the standard used by most software:
https://semver.org/

We do the same for easi. 
This is more or less the same as you're describing.
Bugfixes increase patch, new features increase minor, breaking increase major.

**Comment by AliceGabriel** [comment]
Hi, great! I like the user perspective of Sebastian's approach, as well as
the proposed staggered release (if it's not too late to weigh in)

On Tue, Aug 30, 2022 at 4:31 AM Sebastian Wolf ***@***.***>
wrote:

> I would suggest that we could make it dependent on the parameter files:
>
>    - If the SeisSol core is not changed (e.g. changes to the
>    documentation or analysis scripts): do not increase (e.g. #589
>    <https://github.com/SeisSol/SeisSol/pull/589>)
>    - If the SeisSol core is changed for a bugfix, but a user doesn't
>    realize these changes: increase patch (e.g. #638
>    <https://github.com/SeisSol/SeisSol/pull/638>)
>    - If a new feature is added to the parameter file: increase minor (e.g
>    #644 <https://github.com/SeisSol/SeisSol/pull/644>)
>    - If a feature is removed from the parameter file: increase major
>    (e.g. #625 <https://github.com/SeisSol/SeisSol/pull/625>)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SeisSol/SeisSol/discussions/656#discussioncomment-3508116>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACHOTRALUE2H77X7CMYDHQTV3XWHRANCNFSM57W55IMQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Prof. Alice-Agnes Gabriel
Earthquake Physics, Institute of Geophysics, Department of Earth and
Environmental Sciences, LMU Munich
www.geophysik.uni-muenchen.de/Members/gabriel

**Comment by Thomas-Ulrich** [comment]
What I don't like is that this should not refrain us for making changes.
E.g. If we realize that one TP parameter should have a different name, then it will trigger 2.0.0.
Then people may think it is a big change, as big as dr/cpp but it is just a renaming slightly clarifying the parameter file. 

Le 14 septembre 2022 04:07:56 GMT-07:00, Lukas Krenz ***@***.***> a écrit :

**Reply by krenzland** [reply]
Sorry, that doesn't make sense. E.g. I'm currently using version 104.x.x of Firefox...

**Comment by Thomas-Ulrich** [comment]
ok, then should we tag some of the merged PR since #625 with a version tag?
E.g. I propose:
https://github.com/SeisSol/SeisSol/pull/663 v1.1.0
https://github.com/SeisSol/SeisSol/pull/670 v1.2.0

**Reply by krenzland** [reply]
Sure, we can do that. In principle, we do not need to do a release for every merged PR. Isn't #670 a bugfix not a new feature though?

**Reply by Thomas-Ulrich** [reply]
The energy output was not implemented for GPUs. It was on purpose, so it was not a bug. So for me, that's a new feature ;-)
